rework of dwi.py

IlGnome opened the pull request:
Changed dwi.py to reactions.py
added headdesk and fetish commands
added more macros for all commands
moved image macros to json file located in /data/

IlGnome committed:
Create reactionmacros.json

Contains reactions macros for reactions.py

IlGnome committed:
Change dwi.py to reactions.py

Adding a whole bunch of extra stuff

IlGnome committed:
rewrote plugin

Plugin now handles DWI, Fetish, Facepalm, and Headdesk

IlGnome committed:
Update reactionmacros.json

formatting change.

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+
+    ]
+
+@hook.on_start()
```
- pep8 guidelines recommend 2 blank lines before and after a function definition, not 1

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+
+
+
+@hook.command('dwi','dealwithit')
```
- pep8 guidelines recommend 2 blank lines before and after a function definition, not 3

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+
+@hook.command('dwi','dealwithit')
+
+def DWI(text, message):
```
- local variable and function names should be snake_case

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+
+@hook.command('fp','facepalm')
+
+def FP(text,message):
```
- local variable and function names should be snake_case
- Missing whitespace after comma

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+
+
+@hook.command('fp','facepalm')
+
```
- pep8 guidelines recommend no blank lines after a function decorator

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+
+
+@hook.command('dwi','dealwithit')
+
```
- pep8 guidelines recommend no blank lines after a function decorator

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+
+
+
+
```
possibly extraneous whitespace

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+
+
+
+
```
- pep8 guidelines recommend 2 blank lines before and after a function definition, not 4

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+@hook.command('fp','facepalm')
+
+def FP(text,message):
+    ''' Expresses your frustration with another user. File located in reactions.py'''
```
- Docstrings should use triple quotes, not triple single-quotes.
- Command docstrings should begin with '- ' if they do not specify any arguments, due to how the core generates the help response

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+
+def FP(text,message):
+    ''' Expresses your frustration with another user. File located in reactions.py'''
+    FacePalmer = text.strip()
```
- local variable and function names should be snake_case

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+    FacePalmer = text.strip()
+
+    message('Dammit {} {}'.format(FacePalmer, random.choice(reactionmacros['Facepalmmacros'])))
+
```
- pep8 guidelines recommend 2 blank lines before and after a function definition, not 1

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+    message('Dammit {} {}'.format(FacePalmer, random.choice(reactionmacros['Facepalmmacros'])))
+
+@hook.command('hd', 'headdesk')
+
```
- pep8 guidelines recommend no blank lines after a function decorator

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+
+@hook.command('hd', 'headdesk')
+
+def HD(nick, message):
```
- local variable and function names should be snake_case

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+    idiot = nick.strip()
+
+    message('{} {}'.format(idiot, random.choice(reactionmacros['HeadDeskMacros'])))
+
```
- pep8 guidelines recommend 2 blank lines before and after a function definition, not 1

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+    message('{} {}'.format(idiot, random.choice(reactionmacros['HeadDeskMacros'])))
+
+@hook.command('fetish', 'tmf')
+
```
- pep8 guidelines recommend no blank lines after a function decorator

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+
+@hook.command('fetish', 'tmf')
+
+def Fetish(nick, message):
```
- local variable and function names should be snake_case

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+@hook.command('fetish', 'tmf')
+
+def Fetish(nick, message):
+    ''' Did some one just mention what your fetish was? Let them know! Code located in reactions.py'''
```
- Docstrings should use triple quotes, not triple single-quotes.
- Command docstrings should begin with '- ' if they do not specify any arguments, due to how the core generates the help response

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+
+def Fetish(nick, message):
+    ''' Did some one just mention what your fetish was? Let them know! Code located in reactions.py'''
+    PersonToShareFetishWith = nick.strip()
```
- local variable and function names should be snake_case

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,63 @@
+@hook.on_start()
+def load_macros(bot):
+    global reactionmacros
+    with codecs.open(os.path.join(bot.data_dir,"reactionmacros.json"), encoding = "utf-8") as macros:
```
- Missing whitespace after comma
- Extra whitespace in keyword specifier

edwardslabs commented:
Please make the suggested changes.

IlGnome committed:
Update reactionmacros.json

Added ONE whole new fetish macro

IlGnome committed:
Pep8ing due to feedback.

I probably screwed it up anyway.

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,48 @@
+@hook.command('dwi', 'dealwithit')
+def deal_with_it(text, message):
+    """Tell some one in the channel to deal with it. File located in reactions.py"""
+    PersonNeedsToDeal = text.strip()
```
Local variables should always be in snake_case not PascalCase

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,48 @@
+        
+@hook.command('dwi', 'dealwithit')
+def deal_with_it(text, message):
+    """Tell some one in the channel to deal with it. File located in reactions.py"""
```
This docstring should probably be
```python
"""<nick> - Tell <nick> to deal with it"""
```

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,48 @@
+
+@hook.command('fp', 'facepalm')
+def face_palm(text,message):
+    """Expresses your frustration with another user. File located in reactions.py"""
```
This docstring should be
```python
"""<nick> - Express your frustration with <nick>"""
```

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,48 @@
+
+
+@hook.command('fp', 'facepalm')
+def face_palm(text,message):
```
Missing space after comma

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,48 @@
+@hook.command('fp', 'facepalm')
+def face_palm(text,message):
+    """Expresses your frustration with another user. File located in reactions.py"""
+    FacePalmer = text.strip()
```
Local variables should be snake_case not PascalCase

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,48 @@
+@hook.command('fetish', 'tmf')
+def my_fetish(nick, message):
+    """Did some one just mention what your fetish was? Let them know! Code located in reactions.py"""
+    PersonToShareFetishWith = nick.strip()
```
Local variables should be snake_case not PascalCase

linuxdaemon replied:
Using `nick` here instead of `text` will result in:
```
<User1> {some fetish}
<User2> .tmf User1
<Bot> User2 {reaction}
```
This seems to not be the desired behavior

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,48 @@
+@hook.command('hd', 'headdesk')
+def head_desk(nick, message):
+    """Hit your head against the desk becausae of a user. Code located in reactions.py"""
+    idiot = nick.strip()
```
Same issue as `my_fetish`

linuxdaemon reviewed (commented):
noticed a few possible bugs too

IlGnome committed:
More pep8 stuff

Because linuxdaemon won't stop yelling at me.

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,48 @@
+@hook.command('fetish', 'tmf')
+def my_fetish(text, message):
+    """Did some one just mention what your fetish was? Let them know! Code located in reactions.py"""
+    person_to_share_fetish_with = nick.strip()
```
The variable `nick` is not defined at this point.

IlGnome committed:
commiting more pep8 changes

I hope linuxdaemon chokes on their pizza. Not like chokes to death, but that annoying 'down the wrong pipe' kind of choke.

IlGnome committed:
changed names from PascalCase to snake_case

It was bothering me

IlGnome committed:
More clean up

I hate my life.

IlGnome committed:
Fixed formatting

Now I hate more than my life.

IlGnome committed:
Removed extra white space

The more commits I make, the more violent I tend to become. I know why most coders drink heavily now.

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,47 @@
+def deal_with_it(text, message):
+    """<nick>- Tell <nick> in the channel to deal with it. Code located in reactions.py"""
+    person_needs_to_deal = text.strip()
+    message('{} {}'.format(random.choice(deal_with_it_phrases).format(person_needs_to_deal), random.choice(reaction_macros['deal_with_it_macros'])))
```
This should really not be a single line. Formatting the selection from `deal_with_it_phrases` should be its own line

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,47 @@
+        
+@hook.command('dwi', 'dealwithit')
+def deal_with_it(text, message):
+    """<nick>- Tell <nick> in the channel to deal with it. Code located in reactions.py"""
```
The convention with other plugins is 
```python
"""<nick> - help text"""
```
(note the space before the hyphen)

IlGnome committed:
More code clean up.

Now I'm just doing it to pad my stats.

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,49 @@
+deal_with_it_phrases = [ 'Stop complaining, \x02{}\x02, and',
+               'Jesus fuck \x02{}\x02, just',
+               'Looks like \x02{}\x02 needs to',
+               'Ever think that \x02{}\x02 just needs to']
```
Perhaps moving this to the json file would make future updates easier

linuxdaemon replied:
Wrong continued indentation (add 10 spaces).
```
               'Ever think that \x02{}\x02 just needs to']
               ^         |
```

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,49 @@
+from cloudbot import hook
+
+
+deal_with_it_phrases = [ 'Stop complaining, \x02{}\x02, and',
```
No space allowed after bracket
```
deal_with_it_phrases = [ 'Stop complaining, \x02{}\x02, and',
                       ^
```

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,49 @@
+
+
+deal_with_it_phrases = [ 'Stop complaining, \x02{}\x02, and',
+               'Jesus fuck \x02{}\x02, just',
```
Wrong continued indentation (add 10 spaces).
```
               'Jesus fuck \x02{}\x02, just',
               ^         |
```

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,49 @@
+
+deal_with_it_phrases = [ 'Stop complaining, \x02{}\x02, and',
+               'Jesus fuck \x02{}\x02, just',
+               'Looks like \x02{}\x02 needs to',
```
Wrong continued indentation (add 10 spaces).
```
               'Looks like \x02{}\x02 needs to',
               ^         |
```

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,49 @@
+    with codecs.open(os.path.join(bot.data_dir, "reaction_macros.json"), encoding="utf-8") as macros:
+        reaction_macros = json.load(macros)
+
+        
```
Trailing whitespace

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,49 @@
+    face_palmer = text.strip()
+    message('Dammit {} {}'.format(FacePalmer, random.choice(reaction_macros['facepalm_macros'])))
+
+    
```
Trailing whitespace

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,49 @@
+    idiot = text.strip()
+    message('{} {}'.format(idiot, random.choice(reaction_macros['head_desk_macros'])))
+
+    
```
Trailing whitespace

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,49 @@
+
+@hook.on_start()
+def load_macros(bot):
+    global reaction_macros
```
Global variable 'reaction_macros' undefined at the module level

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,49 @@
+@hook.command('dwi', 'dealwithit')
+def deal_with_it(text, message):
+    """<nick> - Tell <nick> in the channel to deal with it. Code located in reactions.py"""
+    person_needs_to_deal = text.strip()
```
Unused variable 'person_needs_to_deal'

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,49 @@
+def face_palm(text, message):
+    """<nick> - Expresses your frustration with <Nick>. Code located in reactions.py"""
+    face_palmer = text.strip()
+    message('Dammit {} {}'.format(FacePalmer, random.choice(reaction_macros['facepalm_macros'])))
```
Undefined variable 'FacePalmer'

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,49 @@
+@hook.command('fp', 'facepalm')
+def face_palm(text, message):
+    """<nick> - Expresses your frustration with <Nick>. Code located in reactions.py"""
+    face_palmer = text.strip()
```
Unused variable 'face_palmer'

linuxdaemon commented on plugins/reactions.py:
```diff
@@ -0,0 +1,49 @@
+import codecs
+import json
+import asyncio
```
Unused import asyncio

IlGnome committed:
More pep8 stuff

I swear linuxdeamon is just picking on me. That or I'm /really/ bad at coding.
